compiler/rustc_codegen_rmc/src/codegen/function.rs

```diff
@@ -158,30 +148,49 @@ impl<'tcx> GotocCtx<'tcx> {
             ),
         };
 
-        // Now that we have the tuple, write the individual component locals
-        // back to it as a GotoC struct.
+        // When we codegen the function signature elsewhere, we will codegen the untupled version.
```

Comment:
Let me see if I understand, the fix here is to use a BTreeMap to sort the arguments according to their name instead of the order of occurrence.

If that is the case, can you please clarify that in the comment inside the code? In the example you posted, both positional or sorted arguments would yield the same result.

Reply:
Done